hr_timesheet_begin_end/models/account_analytic_line.py

```diff
@@ -15,16 +16,37 @@ class AccountAnalyticLine(models.Model):
     time_start = fields.Float(string="Begin Hour")
     time_stop = fields.Float(string="End Hour")
 
-    @api.constrains("time_start", "time_stop", "unit_amount")
-    def _check_time_start_stop(self):
+    # Override to be a computed field.
+    unit_amount = fields.Float(
+        compute="_compute_unit_amount",
+        store=True,
+        readonly=False,
+    )
+
+    @api.model
+    def _add_unit_amount_to_vals_list(self, vals_list):
+        for vals in vals_list:
+            if not vals.get("unit_amount") and (time_start := vals.get("time_start")):
+                vals["unit_amount"] = self.unit_amount_from_start_stop(
+                    time_start, vals.get("time_stop", 0)
+                )
```

Comment:
I'm not sure this if-statement is valid.

- unit_amount can be 0. Should probably be `vals.get("unit_amount") is None`
- 0 is a valid `time_start` value which is not truthy…

Reply:
no longer valid.